torch/csrc/jit/runtime/static/memory_planner.h

```diff
@@ -199,8 +192,25 @@ class MemoryPlanner {
   // We don't have any guarantee that the model doesn't change the
   // Storage for managed tensors out from under us during execution,
   // so we have to check the StorageImpls each time we deallocate.
-  std::vector<std::pair<size_t, at::StorageImpl>>
-      managed_tensor_storage_impls_{};
+
+  // We will use placement-new to add new storages to this buffer of bytes.
+  alignas(at::StorageImpl) unsigned char* storages_buffer_;
+
+  // This pointer just points to the storage buffer with the proper type
+  at::StorageImpl* storages_;
+
+  // Contains the size (in bytes) of the data to be allocated for each storage
+  size_t* storages_nbytes_;
+
+  // Current number of storages that have been placed into the storage buffer
+  size_t storages_size_;
+
+  // Total capacity of the storage buffer
+  size_t storages_capacity_;
+
+  // This must be called by the destructor of any subclass, so that the buffer
+  // of storages is deleted properly
+  void deleteStorages();
```

Comment:
Maybe it would be better to put this storage buffer stuff into a separate class (maybe called `StorageImplsBuffer`) so that the logic for allocating and deleting is separate, and we wouldn't have to explicitly call this `deleteStorages` function

Reply:
Yeah, a mini wrapper class is safer and highly recommended.